oligo_designer_toolsuite/oligo_specificity_filter/_filter_base.py

```diff
@@ -79,22 +94,48 @@ class AlignmentSpecificityFilter(SpecificityFilterBase):
 
     :param dir_output: Directory for saving intermediate files generated during the filtering process.
     :type dir_output: str
+    :param ai_filter: The machine learning model used to filter the oligos {None, 'hybridization_probability'}, defaults to None
+    :type ai_filter: str, optional
+    :param ai_filter_thershold: The threshold below which the oligos are filtered, defaults to None
+    :type ai_filter_thershold: float, optional
+    :param ai_filter_path: The path to the machine learning model used to filter the oligos, if None the pretrained model provided will be used, defaults to None
+    :type ai_filter_path: str, optional
     """
 
-    def __init__(self, dir_output: str = "output"):
+    def __init__(
+            self, 
+            dir_output: str = "output",
+            ai_filter: str = None,
+            ai_filter_threshold: float = None,
+            ai_filter_path: str = None,
+    ):
         """Construnctor for the AlignmentSpecificityFilter class."""
         # folder where we write the intermediate files
         self.dir_output = dir_output
         Path(self.dir_output).mkdir(parents=True, exist_ok=True)
 
+        if ai_filter is not None and ai_filter_threshold is  None:
+            raise ValueError(
+                "ai_filter_threshold must be specified if ai_filter is specified."
+            )
+        # create ai API
+        self.ai_filter_threshold  = ai_filter_threshold
+        if ai_filter is None:
+            self.ai_filter_api = None
+        else:
+            self.ai_filter_api = generate_api(
```

Comment:
the generate_api function could for instance go into the AI filter class (similar to the policiy classes we have)